compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -4174,12 +4181,12 @@ object Types {
       // We need to invalidate the cache when the period changes because the
       // case `TermRef` of `Type#isStable` reads denotations, which depend on
       // the period. See docs/_docs/internals/periods.md for more information.
-      if myIsStablePeriod != ctx.period then
+      if myisStableRunId != ctx.runId then
```

Comment:
Looks like the comment needs to be updated to reflect the code, but I don't understand why this is correct since `StableRealizable` is not a `FromStartFlags` or `AfterLoadFlags`.

Reply:
`StableRealizable` is computed on completion, and the critical window is covered by `is(Provisional)`.